File: backend/src/Municipal.Domain/ServiceRequest.cs
namespace Municipal.Domain;

public class ServiceRequest
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public RequestCategory Category { get; set; }
    public RequestStatus Status { get; set; } = RequestStatus.Submitted;

    public string Title { get; set; } = "";
    public string Description { get; set; } = "";

    public string? Address { get; set; }
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }

    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset? UpdatedAt { get; set; }
}

File: frontend/src/types/serviceRequest.ts
export type RequestStatus = 0 | 1 | 2 | 3; // align to backend enum values
export type RequestCategory = 0 | 1 | 2 | 3 | 4; // align to backend enum values

export interface ServiceRequest {
  id: string;
  title: string;
  description: string;
  address?: string | null;
  status: RequestStatus;
  category: RequestCategory;
  createdAt: string;
}

export interface CreateServiceRequest {
  title: string;
  description: string;
  address?: string | null;
  category: RequestCategory;
}

export interface PagedResult<T> {
  items: T[];
  page: number;
  pageSize: number;
  total: number;
}
